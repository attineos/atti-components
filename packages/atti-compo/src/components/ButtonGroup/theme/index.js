export default ({ borders, colors }) => ({
  borders: {
    borderWidth: borders.widths.bw2,
    borderRadius: borders.radius.br3,
  },
  colors: {
    border: colors.extraDarkSnow,
  },
})
