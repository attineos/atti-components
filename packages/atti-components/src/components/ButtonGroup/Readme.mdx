---
path: '/buttongroup'
name: 'ButtonGroup'
title: 'ButtonGroup component'
---

Button group :

Group buttons, such as in pagination or pageSize elements.

```js
<ButtonGroup>
  <Button>Prev</Button>
  <Button>1</Button>
  <Button>2</Button>
  <Button>3</Button>
  <Button>Next</Button>
</ButtonGroup>
```
