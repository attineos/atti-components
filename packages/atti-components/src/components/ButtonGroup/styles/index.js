import StyledButtonGroup from './StyledButtonGroup'

export { StyledButtonGroup }
