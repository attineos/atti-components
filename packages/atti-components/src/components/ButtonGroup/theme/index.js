export default ({ borders, colors }) => ({
  borders: {
    borderWidth: borders.widths.bw2,
    borderRadius: borders.radius.medium,
  },
  colors: {
    border: colors.extraDarkSnow,
  },
})
