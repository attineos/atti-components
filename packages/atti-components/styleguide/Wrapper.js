import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
