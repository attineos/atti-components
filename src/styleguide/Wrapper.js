import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
