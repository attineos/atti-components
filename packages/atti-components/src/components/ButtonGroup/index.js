import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { StyledButtonGroup } from './styles'

/**
 * A Simple Button.
 *
 */
const ButtonGroup = ({ children, ...rest }) => (
  <StyledButtonGroup {...rest}>{children}</StyledButtonGroup>
)

ButtonGroup.propTypes = {
  /**
   * The Buttons that will be grouped.
   */
  children: PropTypes.any.isRequired,
}

/** @component */
export default styled(ButtonGroup)``
