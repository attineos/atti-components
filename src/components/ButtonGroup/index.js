import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { StyledButtonGroup } from './styles'
/**
 * A Simple Button.
 *
 */
const ButtonGroup = ({ className, children }) => (
  <StyledButtonGroup className={className}>{children}</StyledButtonGroup>
)

ButtonGroup.defaultProps = {
  className: '',
}

ButtonGroup.propTypes = {
  /**
   * Classes of the ButtonGroup.
   */
  className: PropTypes.string,
  /**
   * The Buttons that will be grouped.
   */
  children: PropTypes.any.isRequired,
}

/** @component */
export default styled(ButtonGroup)``
