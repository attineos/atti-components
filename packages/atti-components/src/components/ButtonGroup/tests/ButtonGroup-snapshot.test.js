/* This file was generated automatically, please do not write in it. */
import React from 'react'
import ButtonGroup from '..'
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer'

const mock = () => {}

it('ButtonGroup renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <ButtonGroup className="Atticomponents">Atticomponents</ButtonGroup>
      </Wrapper>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
