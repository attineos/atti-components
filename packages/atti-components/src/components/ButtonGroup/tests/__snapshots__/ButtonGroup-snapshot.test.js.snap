// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<span
  className="atti-box"
>
  <div
    className="StyledButtonGroup-ze25zo-0 kIA-dXJ ButtonGroup-sc-1lkj4qz-0 bNnEOn Atticomponents"
  >
    Atticomponents
  </div>
</span>
`;
